zxcvbn-cs/Matcher/DateMatcher.cs

```diff
@@ -0,0 +1,312 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Zxcvbn.Matcher
+{
+    /// <summary>
+    /// <para>This matcher attempts to guess dates, with and without date separators. e.g. 1197 (could be 1/1/97) through to 18/12/2015.</para>
+    /// 
+    /// <para>The format for matching dates is quite particular, and only detected years in the range 00-99 and 1900-2019 are considered by
+    /// this matcher.</para>
+    /// </summary>
+    public class DateMatcher : IMatcher
+    {
+        // TODO: This whole matcher is a rather messy but works (just), could do with a touching up. In particular it does not provide matched date details for dates without separators
```

Comment:
Indeed, this class is complete garbage. Remove this comment and add an issue to the repo. You don't need to clean this mess, it's okay.

Reply:
Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/pull/1/commits/feb326236c456372caeffbe65b5ac0c4bf780d96